### PR TITLE
feat: Sync mainnet with testnet

### PR DESCRIPTION
**Pull request by filipdjokic**

### CODE REVIEW

File: cmd/parse/gov/proposal.go [hunk 1]
```diff
@@ -110,7 +114,7 @@ func refreshProposalDetails(parseCtx *parser.Context, proposalID uint64, govModu
 
 	// Handle the MsgSubmitProposal messages
 	for index, msg := range tx.GetMsgs() {
-		if _, ok := msg.(*govtypes.MsgSubmitProposal); !ok {
+		if _, ok := msg.(*govtypesv1beta1.MsgSubmitProposal); !ok {
```
Comment (ankurdotb) [inline cmd/parse/gov/proposal.go:117]: Check on testnet and upstream whether this is accurate...one of these references `govtypesv1beta` and the other references `govtypesv1`

File: modules/bank/source/remote/source.go [hunk 2]
```diff
@@ -28,19 +29,22 @@ func NewSource(source *remote.Source, bankClient banktypes.QueryClient) *Source
 }
 
 // GetBalances implements bankkeeper.Source
-func (s Source) GetBalances(addresses []string, height int64) ([]types.AccountBalance, error) {
+func (s Source) GetBalances(addresses []string, height int64) ([]types.NativeTokenAmount, error) {
 	ctx := remote.GetHeightRequestContext(s.Ctx, height)
 
-	var balances []types.AccountBalance
+	var balances []types.NativeTokenAmount
```
Comment (ankurdotb) [inline modules/bank/source/remote/source.go:35]: Seems like a big/weird change so let's crosscheck if bad merge or same in testnet/upstream

File: modules/gov/expected_modules.go [hunk 3]
```diff
@@ -26,6 +26,7 @@ type SlashingModule interface {
 
 type StakingModule interface {
 	GetStakingPool(height int64) (*types.Pool, error)
+	GetStakingPoolSnapshot(height int64) (*types.PoolSnapshot, error)
```
Comment (ankurdotb) [inline modules/gov/expected_modules.go:29]: On testnet, I saw there were swaps between Pool to PoolSnapshot, this keeps both.

File: modules/gov/source/source.go [hunk 4]
```diff
@@ -1,12 +1,15 @@
...
-	Proposal(height int64, id uint64) (*govtypes.Proposal, error)
-	ProposalDeposit(height int64, id uint64, depositor string) (*govtypes.Deposit, error)
-	TallyResult(height int64, proposalID uint64) (*govtypes.TallyResult, error)
-	DepositParams(height int64) (*govtypes.DepositParams, error)
-	VotingParams(height int64) (*govtypes.VotingParams, error)
-	TallyParams(height int64) (*govtypes.TallyParams, error)
+	Proposal(height int64, id uint64) (govtypesv1beta1.Proposal, error)
+	ProposalDeposit(height int64, id uint64, depositor string) (*govtypesv1.Deposit, error)
+	TallyResult(height int64, proposalID uint64) (*govtypesv1.TallyResult, error)
+	DepositParams(height int64) (*govtypesv1.DepositParams, error)
```
Comment (ankurdotb) [inline modules/gov/source/source.go:12]: Both types needed?

File: modules/bank/source/remote/source.go (hunk 2 above)
Comment (filipdjokic) [reply modules/bank/source/remote/source.go:35]: Same as in the upstream [chains/cheqd/mainnet](https://github.com/forbole/bdjuno/blob/chains/cheqd/mainnet/modules/bank/source/remote/source.go) branch.
But on `cosmos/v0.46.x `it's still `AccountBalance`, so I'm not 100% sure.
On testnet explorer, we used `chains/cheqd/mainnet` as the upstream branch.

File: cmd/parse/gov/proposal.go (hunk 1 above)
Comment (filipdjokic) [reply cmd/parse/gov/proposal.go:117]: This is fine on both the testnet and upstream.

File: modules/gov/expected_modules.go (hunk 3 above)
Comment (filipdjokic) [reply modules/gov/expected_modules.go:29]: Both are kept on the testnet and upstream.

File: modules/gov/source/source.go (hunk 4 above)
Comment (filipdjokic) [reply modules/gov/source/source.go:12]: Looks like it - it was a part of [this commit](https://github.com/forbole/bdjuno/commit/901a355b0b5e2d195f65936c8743ea489ee18766).

**Pull request merged**